### ISSUE TITLE
Feature request: m1 homebrew installation

### ISSUE THREAD

**Issue by Adam2Marsh** [weprovide/valet-plus#568]
- [x] I've checked the issue queue and could not find anything similar to my feature.
- [x] I'm on the latest version of valet-plus (`valet --version`): `Valet+ 2.1.0`

**Why implement this feature?**  
Raising as a feature as couldn't see anyone else trying this and no documents to suggest it's possible. When installing valet plus on a m1 mac with homebrew installed in the recommended `/opt/homebrew` it doesn't work; would be nice for valet to support this.

**What will be required for this feature?**  
I think would need a flag on install to point towards a different installation directory and maybe saved to a config file so future commands work from a new directory.

**Sources**
https://docs.brew.sh/Installation

**Comment by leewillis77** [comment]
I can confirm that the standard installation doesn't work on a brand new M1 MacBook Pro. It fails to detect the installed PHP.

**Comment by leewillis77** [comment]
Note: This might be related to #566 also

**Comment by Adam2Marsh** [comment]
Hi @leewillis77 I've got a fork of the project where I'm working through the issues. Current problem is with a php module which I can't seem to get to compile correctly on my mac, see my stackoverflow here asking for some help - https://stackoverflow.com/questions/66861185/unable-to-compile-php-module-apcu-on-m1-mac

**Comment by Adam2Marsh** [comment]
Update, I have a working version (mostly....). I still need to check all tests and double check a few things but it should work with only 1 extra step I don't have a solution for yet.

If you're using a php version higher than 7.2 you'll need to set the following ENV variables and then install apcu yourself.
```
export LDFLAGS="-L/opt/homebrew/opt/pcre2/lib"
export CPPFLAGS="-I/opt/homebrew/opt/pcre2/include"
export PKG_CONFIG_PATH="/opt/homebrew/opt/pcre2/lib/pkgconfig"
```

Plan this week to tidy it up and see what tests I've broke but if you want to have a look at what I've done and give it a test run please do :) 

Edit:
Removed the steps as they were not good enough, I have started a pull request which has more information on what's added and what's missing as a better place to document incase people want to try.

**Comment by raptor235** [comment]
Hey @Adam2Marsh great work and THANK YOU.. I just got a new M1 and trying valet for the first time running into all these issues... 

With this PR not being merged in yet how would I go about testing your PR on my end? I would like to run through the process without the package being published yet through homebrew...

**Comment by raptor235** [comment]
@mischabraam any possibility of merging this in?

**Comment by Adam2Marsh** [comment]
Hi @raptor235 

The best wasy would be to grab my fork of the project which has the changes, you can find here - https://github.com/Adam2Marsh/valet-plus

There are a couple of issues, please take a look at https://github.com/weprovide/valet-plus/pull/573 as documents some of them.

If you could give it a go, that would be great! Haven't had a chance to wipe my machine and try a fresh since I got it working. 1 thing to note is make sure your terminal is running in the right OS mode, type `arch` to check it's arm. I lost a week as the terminal I was in showed i383 and confused the hell out of me.

It's not ready for merge because the tests are failing on a current mac, I need to go and fix it up. Hoping to get back on it over next couple of weeks to finish it all off and get it merged all in :)

**Comment by peterkeijsers** [comment]
Extra information: The main reason most Valet dependencies break is because Homebrew decided to install the packages in `/opt/homebrew/` when using a M1 Apple Silicon chip. The reasons for that are discussed here: https://github.com/Homebrew/brew/issues/9177. On Intel Chips they install packages in `/usr/local/`. The whole Valet core is filled with hard-coded references to this `/usr/local` directory.

I've changed these references and got it working. But maybe (haven't tried) the quickest solution is to change the Homebrew configuration `HOMEBREW_PREFIX` value back from `/opt/homebrew` to `/usr/local`. And then install valet using `valet install`. But on the long term this isn't a valid solution as the Homebrew core team decided to switch permanently to `/opt/homebrew`. So Adjusting Valet to be compatible with this variable `HOMEBREW_PREFIX` config value is a better option. Use the `brew --prefix` command to read out this value and use it instead of the hardcoded constants in Valet's core php scripts.

**Comment by mattlooby22** [comment]
what a legend @Adam2Marsh . I followed the Valet + install instructions and just before running `valet install` i copied your forked version of Valet repo over the one in ~/.composer/vendor/weprovide. 

Then i ran `valet install --brew-opt` 

Worked first time. Thanks so much

**Comment by bashizee** [comment]
install vallet plus in m1 mac 
installed based on @Adam2Marsh ,

step 1: ##### install brew #####

/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
brew update && brew upgrade

brew tap henkrehorst/php
brew install composer

step 2: ##### select php version you required #####

brew install valet-php@7.4
brew link --force --overwrite valet-php@7.4

step 3: ##### install valet plus #####
composer global require techdivision/valet-plus
grep -q 'export PATH="$PATH:$HOME/.composer/vendor/bin"' ~/.zshrc || echo 'export PATH="$PATH:$HOME/.composer/vendor/bin"' >> ~/.zshrc

#####
this will install valet plus module in /Users/bashid/.composer/vendor/weprovide/
replace this file with file cloned  from https://github.com/Adam2Marsh/valet-plus
#####

valet install --brew-opt
#####
  if this command throws any error ,
   go to this file /Users/bashid/.composer/vendor/weprovide/valet-plus/cli/valet.php then comment  this line (line:37)define('BREW_PATH', Configuration::read()['brewPath']);

  re-run ->  valet install --brew-opt

if you get php error while installing nginx which will appear in red color
  sudo cp /opt/homebrew/Cellar/pcre2/10.38_1/include/pcre2.h  /opt/homebrew/Cellar/valet-php@7.4/7.4.16/include/php/ext/pcre/pcre2.h
  (above path can vary)

  re-run ->  valet install --brew-opt
#####


i hope this will install vallet

**Comment by tebold** [comment]
Hey @mattlooby22 and @bashizee, when I try to run `valet install --brew-opt`, it's telling me `The "--brew-opt" option does not exist`. I can't find any information on this option.

**Comment by mischabraam** [comment]
Released in [2.3.0](https://github.com/weprovide/valet-plus/releases/tag/2.3.0)
